### PR TITLE
ORCH-359 Support new sonar-scanner script with SQ Scanner >= 2.6

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/SonarScanner.java [hunk 1]
```diff
@@ -43,11 +46,28 @@
     return ImmutableMap.of("SONAR_RUNNER_OPTS", "-Djava.awt.headless=true");
   }
 
+  /**
+   * @deprecated since 3.10 use {@link #setScannerVersion(String)}
+   */
   @Override
   public SonarScanner setRunnerVersion(String s) {
```
Comment (SonarTech) [inline sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/SonarScanner.java]: MAJOR Add the missing @Deprecated annotation. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
#### Analysis summary
SonarQube analysis reported 2 issues
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.

File: sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/SonarScanner.java [hunk 2]
```diff
@@ -44,10 +47,33 @@
   }
 
   @Override
+  public SonarScanner setEnvironmentVariable(String name, String value) {
+    return (SonarScanner) super.setEnvironmentVariable(name, value);
+  }
+
+  /**
```
Comment (SonarTech) [inline sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/SonarScanner.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
#### Analysis summary
SonarQube analysis reported 1 issue
* INFO 1 info

Watch the comments in this conversation to review them.

File: sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/SonarScanner.java [hunk 3]
```diff
@@ -44,10 +47,33 @@
...
+   * @deprecated since 3.10 use {@link #setScannerVersion(String)}
+   */
+  @Deprecated
+  @Override
   public SonarScanner setRunnerVersion(String s) {
```
Comment (SonarTech) [inline sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/SonarScanner.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

**Comment by dbmeneses** [comment]
LGTM

**Pull request merged**